### PR TITLE
LPTOCPCI-10: Onboard Fuse Online Operator with UI Tests

### PR DESCRIPTION
**Pull request by amp-rh**
# Summary
At a high level, this PR adds to  `openshift-ci` a configuration that:
- Installs an IPI-based cluster on AWS
- Installs the Fuse Online operator from OperatorHub
- Runs the Fuse online containerized tests
- Archives the test run results

# Proposed Changes:
add syndesisio/syndesis-qe configuration
add test steps to registry for `syndesisio/syndesis-qe` setup and deployment 
add image `fuse-online-test-runner` from `syndesis-qe` base image

closes #[LPTOCPCI-10](https://issues.redhat.com/browse/LPTOCPCI-10)

### CODE REVIEW

File: ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x__fuse-online-ocp4.13-lp-interop.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      stream: nightly
+      version: "4.13"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: fuse-online-interop-aws
+  cron: 0 1 * * 1
```
Comment (chetna14manku) [inline ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x__fuse-online-ocp4.13-lp-interop.yaml:27]:
```suggestion
  cron: 0 6 * * 1
```

File: ci-operator/step-registry/syndesisio/setup-syndesis-qe-env/syndesisio-setup-syndesis-qe-env-commands.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+oc project default || true
+oc registry login || true
+oc create --as system:admin user kubeadmin || true
```
Comment (Simranpal) [inline ci-operator/step-registry/syndesisio/setup-syndesis-qe-env/syndesisio-setup-syndesis-qe-env-commands.sh:9]: Isn't the `kubeadmin` already created

File: ci-operator/step-registry/syndesisio/deploy-syndesis-qe/syndesisio-deploy-syndesis-qe-commands.sh [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      image: >-
+        $FUSE_ONLINE_TEST_RUNNER
+      workingDir: /home/seluser/syndesis-qe
+  volumes:
+    - name: test-run-results
+EOF
+
+oc wait --for=condition=Ready pod/test-runner --timeout=15m || true
+oc logs -c runner -n default -f test-runner || true
+oc wait --for=condition=ContainersReady=false pod/test-runner --timeout=1h30m || true
```
Comment (Simranpal) [inline ci-operator/step-registry/syndesisio/deploy-syndesis-qe/syndesisio-deploy-syndesis-qe-commands.sh:81]: why these statements are set to always true? what happens if the pod creation fails?

File: ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    org: syndesisio
+    repo: syndesis-qe
+  labels:
+    ci-operator.openshift.io/cloud: aws
+    ci-operator.openshift.io/cloud-cluster-profile: aws-cspi-qe
+    ci-operator.openshift.io/variant: fuse-online-ocp4.13-lp-interop
+    ci.openshift.io/generator: prowgen
+    job-release: "4.13"
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws
```
Comment (Simranpal) [inline ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml:19]: You may need to add the [Slack alerts](https://cspi-qe.gitbook.io/ocp-ci-onboarding/tutorials/reporting_guide#how-to-setup-slack-alerts-for-a-scenario) to PQE Slack channel & make sure its public

**Review by Simranpal — COMMENTED** [review]
Please see my comments/questions below. 
I also think that maybe the additional ref creation for syndesis setup could be avoided. There are few oc commands there, which can be moved to deploy execution script.

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws

File: ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml (hunk 4 above)
Comment (amp-rh) [reply ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml:19]: Just [opened a ticket](https://issues.redhat.com/browse/LPTOCPCI-254) to address this. Thanks!

Comment (Simranpal) [reply ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml:19]: Here is also an example: https://github.com/openshift/release/pull/38738/files#diff-47bf248bb2bc930bf70d6c658fea67b33d284b93037c717d4dd254bdd9fe3e9aR22

File: ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
```
Comment (Simranpal) [inline ci-operator/jobs/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.15.x-periodics.yaml:19]: Also add the [gangway-api](https://cspi-qe.gitbook.io/ocp-ci-onboarding/tutorials/triggering_guide#gangway-api) support below job name,  in order to trigger this job outside of the cron if needed.
```suggestion
  name: periodic-ci-syndesisio-syndesis-qe-1.15.x-fuse-online-ocp4.13-lp-interop-fuse-online-interop-aws
    prowjob_defaults:
    tenant_id: gangway-api
```

**Review by Simranpal — COMMENTED** [review]
Please see the failing CI checks:
1.  Add the Openshift-ci-robot & openshift-merge-robot to test GitHub organization . [Requirements](https://cspi-qe.gitbook.io/ocp-ci-onboarding/onboarding/onboarding_guide/prerequisites_guide#github-org-requirements)
2. Seems you have some config file(s) which were not  prefixed correctly. here is the [error](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/39335/pull-ci-openshift-release-master-correctly-sharded-config/1658879918970245120)

File: ci-operator/step-registry/syndesisio/setup-syndesis-qe-env/syndesisio-setup-syndesis-qe-env-commands.sh (hunk 2 above)
Comment (amp-rh) [reply ci-operator/step-registry/syndesisio/setup-syndesis-qe-env/syndesisio-setup-syndesis-qe-env-commands.sh]: Yes, but it is not found by the test suite in the default namespace unless these commands are run.

File: ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.13-lp-interop.yaml
Comment (chetna14manku) [inline ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.13-lp-interop.yaml:1]: @amp-rh we are not testing OCP 4.13 anymore, you can remove this file

File: ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.13-lp-interop.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      stream: nightly
+      version: "4.13"
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: fuse-online-interop-aws
+  cron: 0 6 * * 1
```
Comment (chetna14manku) [inline ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.13-lp-interop.yaml:27]: Please add [gangway](https://cspi-qe.gitbook.io/ocp-ci-onboarding/tutorials/triggering_guide#how-to-add-gangway-api-triggering-to-a-scenario) functionality to support manually trigger for this scenario 
```suggestion
  cron: 0 6 * * 1
  remote_api: true
```

File: ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.14-lp-interop.yaml [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    name: "4.13"
+    namespace: ocp
+    tag: cli
+build_root:
+  image_stream_tag:
+    name: release
+    namespace: openshift
+    tag: golang-1.19
+images:
+- context_dir: docker
```
Comment (ascerra) [inline ci-operator/config/syndesisio/syndesis-qe/syndesisio-syndesis-qe-1.14.x__fuse-online-ocp4.14-lp-interop.yaml:12]: It looks like[ this Dockerfile](https://github.com/syndesisio/syndesis-qe/blob/1.14.x/docker/Dockerfile#L121) ends using entrypoint, when the pod is built I believe that the entrypoint.sh is not being executed.

[A thread ](https://redhat-internal.slack.com/archives/CBN38N3MW/p1682403979742489) from the DPTP basically says that it is ignored.

I would suggest moving anything that is done in the[ entrypoint.sh script](https://github.com/syndesisio/syndesis-qe/blob/1.14.x/docker/entrypoint.sh) and move that to the commands.sh file within the ref to be sure that whatever is supposed to be happening there is actually happening.

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by amp-rh** [comment]
/pj-rehearse periodic-ci-syndesisio-syndesis-qe-1.14.x-fuse-online-ocp4.14-lp-interop-fuse-online-interop-aws

**Comment by smatula** [comment]
@amp-rh You can close this PR. Fuse Online has been on-boarded.
See PR: https://github.com/openshift/release/pull/41247

**Pull request closed without merging**